server/services/generateFixture.js

```diff
@@ -0,0 +1,63 @@
+class TeamNode {
+    // Make a Data structure for a team where they have teamId and adjacency list
+    constructor(teamId) {
+        this.teamId = teamId
+        this.played = []
+    }
+    // Add team to played/adjacency list
+    playTeam(teamNode) {
+        const found = this.played.find((playedTeam) => playedTeam.team.teamId === teamNode.teamId)
+        if (found) found.times += 1
+        else this.played.push({ team: teamNode, times: 1 })
+    }
+    // Find how many games this team has played
+    gamesPlayed() {
+        return this.played
+            .map((playedTeam) => playedTeam.times)
+            .reduce((prev, curr) => prev + curr, 0)
+    }
+    // Private method to find next team to play from adjacency list based on least amount of times played
+    #nextTeamFromPlayedTeams() {
+        return this.played.sort((t1, t2) => t2.times - t1.times)[0].team
+    }
+    // Given an array of Teams, find any team from the Nodes list that this current Node hasen't played
+    #teamsNotPlayed(teams) {
+        return teams.filter(
+            (t) => !this.played.map((playedTeam) => playedTeam.team).includes(t) && t != this
+        )
+    }
+    // Find the next team to play
+    nextTeamToPlay(teams) {
+        const newTeamsToPlay = this.#teamsNotPlayed(teams)
+        if (newTeamsToPlay.length !== 0) return newTeamsToPlay[0]
+        return this.#nextTeamFromPlayedTeams()
+    }
+}
+
+// Returns the TeamNode thats played the least amount of games
+function ejectNextTeamNode(teams) {
+    return teams.sort((t1, t2) => t1.gamesPlayed() - t2.gamesPlayed())[0]
```

Comment:
this should be okay, but good to keep in mind `.sort` is in-place so if the ordering of `teams` is important anywhere else it could potentially mess things up

Reply:
good point, will double check and test more to see if its safe to sort in place

Reply:
actually because this was sorted in place it was the reason why the comment above was sorted differently. Will sort this out, which should also address the comment above